docs/sources/reference/components/otelcol.connector.servicegraph.md

```diff
@@ -61,6 +61,7 @@ Name                        | Type             | Description
 `store_expiration_loop`     | `duration`       | The time to expire old entries from the store periodically.         | `"2s"`       | no
 `metrics_flush_interval`    | `duration`       | The interval at which metrics are flushed to downstream components. | `"0s"`       | no
 `database_name_attribute`   | `string`         | The attribute name used to identify the database name from span attributes. | `"db.name"`  | no
+`virtual_node_peer_attributes` | `list(string)` | List of attributes whose presence in client span will result in creation of a virtual server node | `["peer.service", "db.name", "db.system"]` | no
```

Comment:
```suggestion
`virtual_node_peer_attributes` | `list(string)` | A list of attributes to search for in the client span. A virtual node is created if the attributes are found in the client span. | `["peer.service", "db.name", "db.system"]` | no
```

This description is a bit confusing to the uninitiated.  Is one virtual node created for each attribute? Or a single virtual node is created if any of the attributes are found?

Three attributes are named as "default". Are there any other attributes? How does the user know what these should be? Or are they defined by the user?

Removed `server` form the description for terminology consistency.

Reply:
I agree that it is not very obvious how to use this functionality... unfortunately this is because the OTel docs are not clear about it. I had to go through the code and [past issues](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17196) in order to see why this was added.

We could have this in the table:
```
`virtual_node_peer_attributes` | `list(string)` | A list of attributes to search for in the client span. | `["peer.service", "db.name", "db.system"]` | no
```

And this under the table:

```
`virtual_node_peer_attributes` is useful when an OTel-instrumented client sends a request to a service which is not OTel-instrumented.
Normally, `otelcol.connector.servicegraph` wouldn't be able to pair the client span with a service span, 
because no service span will be received if the service is not OTel-instrumented.
When an edge expires, `otelcol.connector.servicegraph` checks if it has peer attributes listed in `virtual_node_peer_attributes`.
If an attribute is found, the metrics are then aggregated with a virtual node.
```

We should probably add a note like this somewhere in the docs:

```
If no client span is found and `virtual_node_peer_attributes` is not an empty list,
then the service span will be paired with a virtual node called `client="user"`.
This can be useful when a client which is not OTel-instrumented (like a web browser) sends a request to an OTel-instrumented service. 
Without a virtual node, normally the client span will be missing, and the server span will expire without being paired.
```

It's a bit weird that `virtual_node_peer_attributes` has an influence over `client="user"` given that `virtual_node_peer_attributes` is not actually needed for this use case. It'd be worth raising an issue in the upstream OTel repo about this...

@mapno since you seem to be familiar with this feature, do you think this is a good description? I suppose we wouldn't get any "error" and "duration" metrics from this?

Reply:
This makes a LOT more sense with the additional information.

Reply:
Oops, forget that merging suggested change will resolve threads 🫣

Reply:
There we go, I added Paulin's suggested doc. PTAL folks!

Reply:
@ptodev your description is spot-on 👍